lib/datadog/appsec/assets/waf_rules/recommended.json

```diff
@@ -1,9 +1,34 @@
 {
   "version": "2.2",
   "metadata": {
-    "rules_version": "1.3.1"
+    "rules_version": "1.4.1"
   },
   "rules": [
+    {
+      "id": "blk-001-001",
+      "name": "Block IP Addresses",
```

Comment:
(Not for this PR)

I just noticed that the yaml variants of these files are actually a bit smaller:

```
-rw-r--r--  1 ivo.anjo staff 194K Oct 18 11:36 recommended.json
-rw-r--r--  1 ivo.anjo staff 149K Oct 18 11:36 recommended.yaml
-rw-r--r--  1 ivo.anjo staff  75K Oct 18 11:36 risky.json
-rw-r--r--  1 ivo.anjo staff  66K Oct 18 11:36 risky.yaml
-rw-r--r--  1 ivo.anjo staff  72K Oct 18 11:36 strict.json
-rw-r--r--  1 ivo.anjo staff  57K Oct 18 11:36 strict.yaml
```

I wonder if it would make sense to adopt these yaml variants in the future, given Ruby ships a built-in yaml parser as well, so we would not need to add any weird dependencies.

Reply:
Interesting, I did not notice that. I'll consider it.

That said, I'm wary of some consequences of switching to YAML, notably around typing; we previously had some issues with e.g `true` vs `"true"` in some contexts that resulted in an important rule being ignored. AFAIK the JSON ones are more battle-tested simply by virtue of more people using it.

Nonetheless, adding an internal task to keep tabs on this and evaluate.